examples/private_set_intersection/module/rust/src/lib.rs

```diff
@@ -38,105 +38,98 @@ pub mod proto {
-    /// this allows multiple sets of clients to compute their own intersections, also explain what
-    /// the security characteristics are
-    sets: HashMap<String, SetIntersection>,
-}
-
-oak::entrypoint_command_handler!(handler => PrivateSetIntersectionDispatcher<Handler>);
-
-impl PrivateSetIntersection for Handler {
-    fn submit_set(&mut self, req: SubmitSetRequest) -> grpc::Result<()> {
-        let mut current_set = self.sets.entry(req.set_id).or_default();
-
-        if current_set.locked {
-            return Err(grpc::build_status(
-                grpc::Code::FailedPrecondition,
-                "Set contributions are no longer available",
-            ));
-        }
+impl Default for Router {
+    fn default() -> Self {
+        let public_key_label = confidentiality_label(web_assembly_module_signature_tag(
+            &base64::decode(PUBLIC_KEY_BASE64.as_bytes())
+                .expect("Couldn't decode Base64 public key"),
+        ));
+        let handler_command_sender = oak::io::entrypoint_node_create::<
+            PrivateSetIntersectionDispatcher<Handler>,
+        >("handler", &public_key_label, "app")
+        .expect("Couldn't create handler node");
 
-        if current_set.set_count < SET_THRESHOLD {
-            let submitted_set = req.values.iter().cloned().collect::<HashSet<_>>();
-            current_set.values = if current_set.values.is_empty() {
-                submitted_set
-            } else {
-                current_set
-                    .values
-                    .intersection(&submitted_set)
-                    .cloned()
-                    .collect()
-            };
-            current_set.set_count += 1;
-            Ok(())
-        } else {
-            Err(grpc::build_status(
-                grpc::Code::FailedPrecondition,
-                "Maximum number of contributed private sets is reached",
-            ))
+        Self {
+            handler_command_sender,
+            public_key_label,
         }
     }
+}
+
+impl oak::CommandHandler for Router {
+    type Command = oak::grpc::Invocation;
 
-    fn get_intersection(
-        &mut self,
-        req: GetIntersectionRequest,
-    ) -> grpc::Result<GetIntersectionResponse> {
-        match self.sets.get_mut(&req.set_id) {
-            Some(ref mut set) => {
-                set.locked = true;
-                Ok(GetIntersectionResponse {
-                    values: set.values.iter().cloned().collect(),
-                })
+    fn handle_command(&mut self, command: oak::grpc::Invocation) -> anyhow::Result<()> {
+        match (&command.receiver, &command.sender) {
+            (Some(receiver), Some(sender)) => {
+                let label = receiver.label()?;
+                let grpc_response_writer = oak::grpc::ChannelResponseWriter::new(sender.clone());
+                // Check if request label is valid in the context of Private Set Intersection
+                // application.
+                if label != self.public_key_label {
```

Comment:
I think in general this should check `flows_to` rather than equality. In particular, if someone wanted to send public data, that should be allowed. WDYT?